### PR TITLE
[Defacepocalypse] De-deface payment methods new and form

### PR DESCRIPTION
**Pull request by HugsDaniel**
#### What? Why?

Part of #3121 

Import payment_methods new view and form partial in our codebase and apply overrides.
I took advantage of this PR to de-deface edit view as well, since it uses the form partial. @luisramos0 @sauloperez what do you think ?

#### What should we test?

In Config -> Payment Methods -> New Payment Method, the form appears as usual.
In Config -> Payment Methods -> Edit, the form appears as usual.

#### Release notes

Import Spree views for `payment_methods` new and edit in our codebase.

Changelog Category: Added

#### How is this related to the Spree upgrade?

This is a first step towards solving #3121.

### CODE REVIEW

File: app/views/spree/admin/payment_methods/edit.html.haml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  %li
+    = button_link_to t(:back_to_payment_methods_list), spree.admin_payment_methods_path, :icon => 'icon-arrow-left'
+= render :partial => 'spree/shared/error_messages', :locals => { :target => @payment_method }
+= form_for @payment_method, :url => admin_payment_method_path(@payment_method) do |f|
+  %fieldset.no-border-top
+    = render :partial => 'form', :locals => { :f => f }
+    .one.column &nbsp;
+    = render :partial => 'spree/admin/shared/hubs_sidebar', :locals => { f: f, klass: :payment_method }
+    .clear
+    .filter-actions.actions{"data-hook" => "buttons"}
```
Comment (luisramos0) [inline app/views/spree/admin/payment_methods/edit.html.haml]: is this data hook needed?

File: app/views/spree/admin/payment_methods/edit.html.haml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+- content_for :page_title do
+  = t(:editing_payment_method)
+  %i.icon-arrow-right
+  = @payment_method.name
+- content_for :page_actions do
+  %li
+    = button_link_to t(:new), spree.new_admin_payment_method_path, :icon => 'icon-plus'
```
Comment (luisramos0) [inline app/views/spree/admin/payment_methods/edit.html.haml]: can we can convert these hashrockets?

File: app/views/spree/admin/payment_methods/edit.html.haml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+- content_for :page_title do
+  = t(:editing_payment_method)
```
Comment (luisramos0) [inline app/views/spree/admin/payment_methods/edit.html.haml]: I'd move these to be lazy lookups if it is not too much trouble.

Comment (HugsDaniel) [reply app/views/spree/admin/payment_methods/edit.html.haml]: What are lazy hookups ?

Comment (luisramos0) [reply app/views/spree/admin/payment_methods/edit.html.haml]: see here:
https://github.com/openfoodfoundation/openfoodnetwork/wiki/Internationalisation-(i18n)#development
Under Add translations
it's using nested structure not root level en.yml entries
basically use t(".editing_payment_method")
and move the en key under en.spree.admin.payment_method

Comment (HugsDaniel) [reply app/views/spree/admin/payment_methods/edit.html.haml]: `ng-app` right ?

File: app/views/spree/admin/payment_methods/edit.html.haml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+- content_for :page_title do
+  = t('.editing_payment_method')
+  %i.icon-arrow-right
+  = @payment_method.name
+- content_for :page_actions do
+  %li
+    = button_link_to t(:new), spree.new_admin_payment_method_path, icon: 'icon-plus'
+  %li
+    = button_link_to t(:back_to_payment_methods_list), spree.admin_payment_methods_path, icon: 'icon-arrow-left'
```
Comment (mkllnk) [inline app/views/spree/admin/payment_methods/edit.html.haml]: This one would be good as lazy lookup as well.

File: app/views/spree/admin/payment_methods/new.html.haml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+- content_for :page_title do
+  = t('.new_payment_method')
+- content_for :page_actions do
+  %li
+    = button_link_to t(:back_to_payment_methods_list), admin_payment_methods_path, icon: 'icon-arrow-left'
```
Comment (mkllnk) [inline app/views/spree/admin/payment_methods/new.html.haml]: Same here.

**Comment by HugsDaniel** [comment]
@luisramos0 looks good ?

**Comment by daniellemoorhead** [comment]
And another one waiting for testing for over a week. Let's get to it, @sauloperez or @luisramos0 or @Matt-Yorkley @mkllnk can you get it onto the next available staging server (currently AU and UK are in use) and get it ready for testing?

#toottoot

**Comment by HugsDaniel** [comment]
Note that this won't close #3121, it's just the first step.

**Comment by RachL** [comment]
Noticed a small issue but pre-existent, so created https://github.com/openfoodfoundation/openfoodnetwork/issues/3225

Testing notes : https://docs.google.com/document/d/1bKwz3sQCvms6wSz_DrDp9DpwErJVHgSEjPXSqt8UTto/edit#

**Pull request merged**